src/ai/backend/manager/models/vfolder.py

```diff
@@ -2558,9 +2559,21 @@ async def build_ctx_in_domain_scope(
     async def build_ctx_in_project_scope(
         self, ctx: ClientContext, scope: ProjectScope
     ) -> VFolderPermissionContext:
+        from .group import AssocGroupUserRow
+
         permission_ctx = await self._build_at_project_scope_non_recursively(ctx, scope.project_id)
-        _user_perm_ctx = await self._build_at_user_scope_non_recursively(ctx, ctx.user_id)
-        permission_ctx.merge(_user_perm_ctx)
+        project_query = sa.select(AssocGroupUserRow).where(
+            AssocGroupUserRow.group_id == scope.project_id
+        )
+        user_perm_ctx = await self._build_at_user_scope_non_recursively(ctx, ctx.user_id)
+        permission_ctx.merge(user_perm_ctx)
+        roles = await get_predefined_roles_in_scope(ctx, scope, self.db_session)
+        if PredefinedRole.ADMIN in roles:
```

Comment:
Is there a reason to do the project query first?
Shouldn't you query the value just before using it?